proof/crefine/RISCV64/Ipc_C.thy

```diff
@@ -5643,7 +5643,7 @@ lemma reply_push_ccorres:
                            simp: typ_heap_simps' creply_relation_def option_to_ctcb_ptr_def
                           split: if_splits)
        apply ceqv
-      apply (ctac add: setThreadStateBlockedOnReply_ccorres)
+      apply (ctac add: setThreadStateBlockedOnReply_ccorres, rename_tac xfdc')
```

Comment:
I don't seem to remember ever doing a `rename_tac xfdc'`, can you say what's happening here?

Reply:
I remember these from the Isabelle update. The name spaces for constants and variables are now separate. That means Isabelle will happily generate the variable name `xfdc` in the goal, which is then inaccessible, because when you type `xfdc` you will get the constant `xfdc` instead. Hence the rename to `xfdc'` which does remain accessible.

Edit:  it's the other way around, I think the bound variable shadows the constant, but you want to be able to use the constant `xfdc` in rule instantiations. In any case, they clash.

Reply:
Yes, this came from the Isabelle update and the edit is exactly right with what is happening. There were a couple of examples in the Isabelle update commit on master doing this `rename xfdc'` which is where I got the idea from. I didn't investigate why it's needed so much more in the MCS proofs though. I guess we explicitly refer to `xfdc` more in them.